### PR TITLE
Fixes output functionality

### PR DESCRIPTION
**Pull request by navidcy**
We need to add tests before merging this.

### CODE REVIEW

**Comment by navidcy** [comment]
`withoutjld2` function errors when `path` is less then 4 characters...

https://github.com/FourierFlows/FourierFlows.jl/blob/84bdb9b4d0864e00cfef3cca2359c1ff7a776610/src/output.jl#L13

**Comment by navidcy** [comment]
`uniquepath` function is not behaving as expected.
For example, if `foo.jld2` and `foo_1.jld2` **both** exist then 

```julia
julia> FourierFlows.uniquepath("foo.jld2")
"for_1_2.jld2"
```

**Comment by navidcy** [comment]
I think that the instance
https://github.com/FourierFlows/FourierFlows.jl/blob/1a8280e037c6d86670d5973661e21dbeb38a6e90/src/output.jl#L32-L34
is never called as the code stands now.

**Comment by navidcy** [comment]
@glwagner, this is ready for review and merge.

File: src/output.jl [hunk 1]
```diff
@@ -57,21 +61,21 @@ end
 
 Saves some parameters of `prob.field`.
 """
-function savefields(file, grid::TwoDGrid)
+function savefields(file::JLD2.JLDFile{JLD2.MmapIO}, grid::TwoDGrid)
```
Comment (glwagner) [inline src/output.jl]: You probably don’t need the type parameter of JLDFile here

File: src/output.jl [hunk 2]
```diff
@@ -10,18 +10,21 @@ struct Output
...
 function uniquepath(path)
-  n = 0
+  n = 1
+  if isfile(path)
+    path = withoutjld2(path) * "_$n.jld2"
+  end
   while isfile(path)
     n += 1
-    path = withoutjld2(path) * "_$n.jld2"
+    path = withoutjld2(path)[1:end-length("_$(n-1)")] * "_$n.jld2"
```
Comment (glwagner) [inline src/output.jl]: 👏

**Pull request merged**